### ISSUE TITLE
Add "NCBI_BSgenome_datapkg" package template to BSgenomeForge and try createPackage() on it

### ISSUE THREAD

**Issue by hpages** [Bioconductor/BSgenomeForge#9]
To do only once PR #8 is merged. Make sure to resync your fork once PR #8 is merged.

Please note that I renamed `makeBSgenomeDataPkgFromNCBI()` -> `forgeBSgenomeDataPkgFromNCBI()` from the initial project proposal. I thought the new name might be slightly better. 

**The `createPackage()` function**

`forgeBSgenomeDataPkgFromNCBI()` is going to use the `createPackage()` function from the **Biobase** package to create the directory structure of the BSgenome data package to forge. Check `?createPackage` in **Biobase** for the details of how this function works.

Note that the "old" `forgeBSgenomeDataPkg()` function from the **BSgenome** package also uses `createPackage()` internally to create the directory structure of the BSgenome data package. The _package template_ used by `forgeBSgenomeDataPkg()` is the `inst/pkgtemplates/BSgenome_datapkg` directory in the **BSgenome** package.

**What you need to do**

1. Carefully read `?createPackage` and make sure that you understand what the function does, how it works, and how to use it.

2. Copy `inst/pkgtemplates/BSgenome_datapkg` from **BSgenome** to the **BSgenomeForge** package, and rename it `inst/pkgtemplates/NCBI_BSgenome_datapkg`. Don't forget to call `git add` on the new `inst` folder.

    The `inst/pkgtemplates/NCBI_BSgenome_datapkg` folder is the _package template_ that we're going to use in `forgeBSgenomeDataPkgFromNCBI()` to create the directory structure of the BSgenome data package to forge. Take a close look at its content. As you can see, the files in it contain a lot of _template symbols_ e.g`@PKGNAME@`, `@PKGTITLE@`, `@ORGANISM@`, etc... If we want to use this template in the context of `createPackage()`, we need to provide a value for each _template symbol_. This is done via the `symbolValues` argument of the `createPackage()` function. But before we do this, we're going to simplify the `NCBI_BSgenome_datapkg` package template a little. The goal of this simplification is to reduce the number of _template symbols_. See below.

5. Simplify the `NCBI_BSgenome_datapkg` template:

    - In `DESCRIPTION`: Remove the `common_name`, `release_date`, and `source_url` lines.

    - In `R/zzz.R`: Replace lines `common_name="@COMMONNAME@",`, `release_date="@RELEASEDATE@",`, and `source_url="@SOURCEURL@",`, with `common_name=NA,`, `release_date=NA,`, and `source_url=NA,`.

    - In `man/package.Rd`: Remove the `\details` and `\note` sections. Remove the `@PKGEXAMPLES@` line in the `\examples` section.

    Note that we will probably make other modifications to this package template once we start working on the `forgeBSgenomeDataPkgFromNCBI()`  function itself.

8. Commit, push, and submit a new PR.

9. Try to call `createPackage()` on the `NCBI_BSgenome_datapkg` template. For this you will need to supply at least 4 arguments (as explained in `?createPackage`): `pkgname`, `destinationDir`, `originDir`, and `symbolValues`. It doesn't really matter what you use for `pkgname` and `symbolValues`, because we are not trying to forge a real BSgenome data package for now. We're only playing around with the `createPackage()` function in order to get familiar with it.

In our next task, we're going to start working on the `forgeBSgenomeDataPkgFromNCBI()`  function itself. We're going to use everything we've done and learned so far for this function.

Let me know if you have questions.

**Comment by kakopo** [comment]
Hi @hpages. 
Thank you for the explanations. As I carried out this task I recalled you had explained a little bit of this during the contribution period, so it was easier to understand and carry out. 
While the task was fairly easy to do for a naive user, I found a few parts of the process redundant, particularly using the `symbolValues` argument, which required a lot of specifications. Wouldn't it be best for some values to be left out optionally in the `symbolValues` list (such as `DESCRIPTION`) and filled in from the template using a text editor?
I have just submitted a PR, although it has a few merge issues that I am trying to resolve. Looking forward to the next task!

**Comment by hpages** [comment]
Yes we could probably simplify the package template a little bit more in order to eliminate a few more symbols like `@PKGDESCRIPTION@`, `@PROVIDER@`, and `@SUGGESTS@`, if that's what you mean. You'll get a better idea of how the package template can be polished/improved once you start working on the `forgeBSgenomeDataPkgFromNCBI()` function itself. See issue #11.

**Comment by hpages** [comment]
What to do with `@PKGDESCRIPTION@`, `@PROVIDER@`, and `@SUGGESTS@` is now discussed in more details in issue #11.
